main.tf

```diff
@@ -707,3 +712,25 @@ resource "aws_s3_bucket_ownership_controls" "this" {
     aws_s3_bucket.this
   ]
 }
+
+resource "aws_s3_bucket_intelligent_tiering_configuration" "tiering_conf" {
+  for_each = local.intelligent_tiering_config
+  name     = each.key
+  bucket   = aws_s3_bucket.this[0].id
+  status   = each.value.status
+
+  filter {
+    prefix = try(each.value.filter["prefix"], "/")
+    tags   = try(each.value.filter["tags"], {})
+  }
+
+  dynamic "tiering" {
+    for_each = each.value.tiering
+
+    content {
+      access_tier = try(tiering.key, null)
```

Comment:
`access_tier` and `days` are required, so there is no need to wrap values with `try()`